jailer.go

```diff
@@ -0,0 +1,376 @@
+	b.secCompLevel = level
+	return b
+}
+
+// Stdout will return the stdout that will be used when creating the
+// firecracker exec.Command
+func (b JailerCommandBuilder) Stdout() io.Writer {
+	return b.stdout
+}
+
+// WithStdout specifies which io.Writer to use in place of the os.Stdout in the
+// firecracker exec.Command.
+func (b JailerCommandBuilder) WithStdout(stdout io.Writer) JailerCommandBuilder {
+	b.stdout = stdout
+	return b
+}
+
+// Stderr will return the stderr that will be used when creating the
+// firecracker exec.Command
+func (b JailerCommandBuilder) Stderr() io.Writer {
+	return b.stderr
+}
+
+// WithStderr specifies which io.Writer to use in place of the os.Stderr in the
+// firecracker exec.Command.
+func (b JailerCommandBuilder) WithStderr(stderr io.Writer) JailerCommandBuilder {
+	b.stderr = stderr
+	return b
+}
+
+// Stdin will return the stdin that will be used when creating the firecracker
+// exec.Command
+func (b JailerCommandBuilder) Stdin() io.Reader {
+	return b.stdin
+}
+
+// WithStdin specifies which io.Reader to use in place of the os.Stdin in the
+// firecracker exec.Command.
+func (b JailerCommandBuilder) WithStdin(stdin io.Reader) JailerCommandBuilder {
+	b.stdin = stdin
+	return b
+}
+
+// Build will build a jailer command.
+func (b JailerCommandBuilder) Build(ctx context.Context) *exec.Cmd {
+	cmd := exec.CommandContext(
+		ctx,
+		"jailer",
+		b.Args()...,
+	)
+
+	fmt.Println("COMMAND", cmd.Args)
+
+	if stdin := b.Stdin(); stdin != nil {
+		cmd.Stdin = stdin
+	}
+
+	if stdout := b.Stdout(); stdout != nil {
+		cmd.Stdout = stdout
+	}
+
+	if stderr := b.Stderr(); stderr != nil {
+		cmd.Stderr = stderr
+	}
+
+	return cmd
+}
+
+// Jail will set up proper handlers and remove configuuration validation due to
+// stating of files
+func jail(ctx context.Context, m *Machine, cfg Config) {
+	m.Handlers.Validation = m.Handlers.Validation.Remove(ValidateCfgHandlerName)
+
+	id := StringValue(cfg.JailerCfg.ID)
+	b := JailerCommandBuilder{}.
+		WithID(id).
+		WithUID(IntValue(cfg.JailerCfg.UID)).
+		WithGID(IntValue(cfg.JailerCfg.GID)).
+		WithNumaNode(IntValue(cfg.JailerCfg.NumaNode)).
+		WithExecFile(StringValue(cfg.JailerCfg.ExecFile)).
+		WithDaemonize(cfg.JailerCfg.Daemonize).
+		WithSecCompLevel(cfg.JailerCfg.SecCompLevel).
+		WithStdout(os.Stdout).
+		WithStderr(os.Stderr)
+
+	rootfs := ""
+	if len(cfg.JailerCfg.ChrootBaseDir) > 0 {
+		rootfs = filepath.Join(cfg.JailerCfg.ChrootBaseDir, "firecracker", id)
+		b = b.WithChrootBaseDir(cfg.JailerCfg.ChrootBaseDir)
+	} else {
+		const defaultJailerPath = "/srv/jailer/firecracker"
+		rootfs = filepath.Join(defaultJailerPath, id)
+	}
+
+	if len(cfg.JailerCfg.NetNS) > 0 {
+		b = b.WithNetNS(cfg.JailerCfg.NetNS)
+	}
+
+	m.cmd = b.Build(ctx)
+	// TODO: hmm, do we actually want to overwrite the socket path?
+	m.cfg.SocketPath = filepath.Join(rootfs, "api.socket")
+
+	m.Handlers.Validation = m.Handlers.Validation.Remove(ValidateCfgHandlerName)
```

Comment:
Added a comment addressing why this was removed:

```go
  // ValidateCfgHandlerName handler needs to be removed due to checking for    
  // non-existant drive paths and socket path. Since the jailer uses relative    
  // links according to its rootfs then the user would also need those files    
  // relative to running this. 
```